packages/models-library/tests/test_user_preferences.py

```diff
@@ -0,0 +1,162 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+from collections.abc import Iterator
+from copy import deepcopy
+from typing import Any
+
+import arrow
+import pytest
+from models_library.services import ServiceKey
+from models_library.user_preferences import (
+    BaseBackendUserPreference,
+    BaseFrontendUserPreference,
+    BaseUserServiceUserPreference,
+    PreferenceType,
+    _AutoRegisterMeta,
+    _BaseUserPreferenceModel,
+)
+from pydantic import parse_obj_as
+
+_SERVICE_KEY_SAMPLES: list[str] = [
+    "simcore/services/comp/something-1231",
+    "simcore/services/dynamic/something-1231",
+    "simcore/services/frontend/something-1231",
+]
+
+
+@pytest.fixture(params=[None, 1, 1.0, "str", {"a": "dict"}, ["a", "list"]])
+def value(request: pytest.FixtureRequest) -> Any:
+    return request.param
+
+
+def _get_base_user_preferences_data(
+    preference_type: PreferenceType, value: Any
+) -> dict[str, Any]:
+    return {"preference_type": preference_type, "value": value}
+
+
+def _get_utc_timestamp() -> float:
+    return arrow.utcnow().datetime.timestamp()
+
+
+@pytest.mark.parametrize("preference_type", PreferenceType)
+def test_base_user_preference_model(value: Any, preference_type: PreferenceType):
+    base_data = _get_base_user_preferences_data(
+        preference_type=preference_type, value=value
+    )
+    assert parse_obj_as(_BaseUserPreferenceModel, base_data)
```

Comment:
should you not check that you get the actual preference you expect? i.e. frontend, backend or service?

Reply:
yes, but this is a union which might create the wrong type if one does not check. that is what I wanted to make you aware of.

Reply:
and also I think Pedro already has a test that just checks examples... if you have any one defined.

Reply:
This is not a union. I think you are referring to `AnyUserPreference` which is not used for any parsing purposes.